diff --git a/msgspec/_core.c b/msgspec/_core.c
index 5239f9f..5f7c99a 100644
--- a/msgspec/_core.c
+++ b/msgspec/_core.c
@@ -5852,7 +5852,7 @@ typedef struct JSONDecoderState {
 
     /* Temporary scratch space */
     unsigned char *scratch;
-    ssize_t scratch_capacity;
+    Py_ssize_t scratch_capacity;
     Py_ssize_t scratch_len;
 
     /* Per-message attributes */
@@ -6437,7 +6437,7 @@ json_decode_array(
 #define JS_SCRATCH_MAX_SIZE 1024
 
 static int
-json_scratch_resize(JSONDecoderState *state, ssize_t size) {
+json_scratch_resize(JSONDecoderState *state, Py_ssize_t size) {
     unsigned char *temp = PyMem_Realloc(state->scratch, size);
     if (MS_UNLIKELY(temp == NULL)) {
         PyErr_NoMemory();
